packages/editor/src/store/actions.js

```diff
@@ -183,8 +182,12 @@ export function* setupEditor( post, edits, template ) {
 		edits,
 		template,
 	};
-	yield resetEditorBlocks( blocks );
+	yield resetEditorBlocks( blocks, { __unstableShouldCreateUndoLevel: false } );
 	yield setupEditorState( post );
+	if ( edits ) {
+		const record = { ...post, ...edits };
+		yield dispatch( 'core', 'receiveEntityRecords', 'postType', post.type, record );
```

Comment:
I have a lot of thoughts here:

 - I'd have thought that we don't need the `post` object to setup the editor anymore. The `id` could have been enough and just call `getEntityRecord` to load the post when needed (selectors...) 
 - It seems like we're considering the "initialEdits" as something "persisted". I believe this is trying to avoid undo/redo levels? 

Basically what's bothering me here is that I think of `receiveEntityRecords` as an internal action to the Core data module that shouldn't be used by other modules. Other modules could say things like `fetch entity` / `edit entity`... but not directly `receive entity`.

Reply:
>It seems like we're considering the "initialEdits" as something "persisted". I believe this is trying to avoid undo/redo levels?

Yes, that's how it worked before.

Reply:
How can we rewrite this to avoid using internal stuff?

Reply:
Why is that a goal? I wouldn't see these as internals. Eventually we should deprecate most of the selectors in the editor store and rely on entities directly.

Things like `editEntityRecord` are already being used here.

Reply:
Because, I think it will allow us to remove a lot of things and will simplify things once we'll have to register multiple entities to edits in the editor store.

I feel like we absorbed everything properly in the core-data module when it . comes to edits except the initial edits, could these be just a call to `editEntityRecord` but just flagging the change as non persistent/non dirty or something like that?

I don't see it as crucial for this PR directly but I think we'd need to handle this properly at some point.

Reply:
>register multiple entities to edits in the editor store

Why would we ever need to do that? Do you mean having one store setting the same edit on two entities? We could just make one entity that internally does that. There's no need to add that to the editor.

>could these be just a call to editEntityRecord but just flagging the change as non persistent/non dirty or something like that?

What's the difference? The edit function is also from Core Data. Why do you see this as a not proper way to do it. This is kind of what the "receive" actions are for; for setting the initial values of a data structure.

Reply:
> Why would we ever need to do that? Do you mean having one store setting the same edit on two entities? We could just make one entity that internally does that. There's no need to add that to the editor.

Maybe not, I don't really know yet. The thing is at some point we'd have to show in the editor what entities were modified... if we do this in a global way, theorically we can have edits lingering from other pages, plugins... which the editor might not be concerned about.

> What's the difference? The edit function is also from Core Data. Why do you see this as a not proper way to do it. This is kind of what the "receive" actions are for; for setting the initial values of a data structure.

Because technically speaking, the initial edits are edits, these are not values you get when calling the REST API which makes me think these values shouldn't be returned when you call `getEntityRecord` for instance. (Imagine a third party plug, a modal, another page calling that selector and receiving values that are different from the real persisted value).

Reply:
>Maybe not, I don't really know yet. The thing is at some point we'd have to show in the editor what entities were modified... if we do this in a global way, theorically we can have edits lingering from other pages, plugins... which the editor might not be concerned about.

That's fine. Receiving an entity does not clear edits if they are different to their persisted counterparts.

>Because technically speaking, the initial edits are edits, these are not values you get when calling the REST API which makes me think these values shouldn't be returned when you call getEntityRecord for instance. (Imagine a third party plug, a modal, another page calling that selector and receiving values that are different from the real persisted value).

What's the use case for initial edits then? Is it just for the "demo" content? Why shouldn't that make the post dirty? I think I would actually prefer initial edits to make the post dirty and just use a regular `editEntityRecord`.

To me this:

>the initial edits are edits, these are not values you get when calling the REST API

Makes me think the entity should show as dirty. An initial edit that doesn't "count as dirty" could overwrite an edit made by another plugin, to the same entity, and remove the dirty state. It would be really confusing.

Reply:
> What's the use case for initial edits then? Is it just for the "demo" content? Why shouldn't that make the post dirty? I think I would actually prefer initial edits to make the post dirty and just use a regular editEntityRecord.

I agree with you here. I think the concern is the drafts that get saved too often maybe when you open the demo content post. But maybe it's fine at this stage of the plugin phase. What do you all think? @jasmussen @mtias Is it ok to consider the "demo" post as dirty as soon as you open it. (This would simplify some things in the code).

Reply:
I would not optimize towards us having the demo content. For one, it is only present in the plugin, and for another, it is intended only as that: demo content to see what the block editor might do. If clicking the demo content button _immediately saves a draft_, that's fine IMO — I think it's primarily us developers who end up with a ton of saved demo content drafts, and even then it's harmless if messy.

Reply:
>If clicking the demo content button immediately saves a draft

It wouldn't do that. It would just load with the post dirty/save button enabled instead of waiting for the first user edit like it does now.

Reply:
> It wouldn't do that. It would just load with the post dirty/save button enabled instead of waiting for the first user edit like it does now.

Isn't dirtiness used a check to start autosaving too (after some seconds) ?

Regardless, it feels like if the "initial edits" are only used for demo right now, we can just remove that behavior and just rely on regular edits.

Reply:
>Isn't dirtiness used a check to start autosaving too (after some seconds) ?

Yes

>Regardless, it feels like if the "initial edits" are only used for demo right now, we can just remove that behavior and just rely on regular edits.

So should I change it and the corresponding e2e tests?

Reply:
let's do iit 👍

Reply:
Done!